UnityProject/Assets/Scripts/Systems/Chemistry/Reaction.cs

```diff
@@ -34,11 +34,20 @@ public virtual bool Apply(object sender, ReagentMix reagentMix)
 		{
 			if (IsReactionValid(reagentMix) == false) return false;
 
-			ApplyReaction(sender as MonoBehaviour, reagentMix);
+			ApplyReaction(sender as GameObject, reagentMix);
 
 			return true;
 		}
 
+		public virtual void OnExposedToHotSpot(GameObject sender, float heat, ReagentMix inMix)
```

Comment:
surely be better to raise the temperature, in the reagent mix and then check if there's any valid reaction?